lua/lspconfig/server_configurations/kotlin_language_server.lua

```diff
@@ -24,6 +24,9 @@ return {
     filetypes = { 'kotlin' },
     root_dir = util.root_pattern(unpack(root_files)),
     cmd = { bin_name },
+    init_options = {
+      storagePath = util.root_pattern(unpack(root_files))(vim.fn.expand('%:p:h')),
```

Comment:
simply api.nvim_buf_get_name(0)

Reply:
I tried to use vim.api.nvim_buf_get_name(0). But root_pattern expects a directory.

When I used `util.root_pattern(unpack(root_files))(vim.api.nvim_buf_get_name(0))` it was not resolving to the correct path somehow. I thought the root pattern function wants a directory(it was used like this in tests) so I changed it to pass the current directory.